avm/ptn/deployment-script/import-image-to-acr/main.bicep

```diff
@@ -0,0 +1,255 @@
+// param rbacRoleNeeded string = '8311e382-0749-4cb8-b61a-304f252e45ec' // AcrPush
+param rbacRoleNeeded string = 'b24988ac-6180-42a0-ab88-20f7382dd24c' //Contributor is needed to import ACR
+
+@description('Optional. Does the Managed Identity already exists, or should be created. Default is false.')
+param useExistingManagedIdentity bool = false
+
+@description('Conditional. Name of the Managed Identity resource to create. Required if `useExistingManagedIdentity` is `true`. Defaults to `id-ContainerRegistryImport`.')
+param managedIdentityName string = 'id-ContainerRegistryImport'
+
+@description('Conditional. For an existing Managed Identity, the Subscription Id it is located in. Default is the current subscription. Required if `useExistingManagedIdentity` is `true`. Defaults to the curent subscription.')
+param existingManagedIdentitySubId string = subscription().subscriptionId
+
+@description('Conditional. For an existing Managed Identity, the Resource Group it is located in. Default is the current resource group. Required if `useExistingManagedIdentity` is `true`. Defaults to the current resource group.')
+param existingManagedIdentityResourceGroupName string = resourceGroup().name
+
+@description('Required. A fully qualified image name to import.')
+@metadata({
+  example: 'mcr.microsoft.com/k8se/quickstart-jobs:latest'
+})
+param image string
+
+@description('Optional. The image will be overwritten if it already exists in the ACR with the same tag. Default is false.')
+param overwriteExistingImage bool = false
+
+@description('Optional. A delay in seconds before the script import operation starts. Primarily to allow Azure AAD Role Assignments to propagate. Default is 30s.')
+param initialScriptDelay int = 30
+
+@description('Optional. The maximum number of retries for the script import operation. Default is 3.')
+param retryMax int = 3
+
+@allowed([
+  'OnSuccess'
+  'OnExpiration'
+  'Always'
+])
+@description('Optional. When the script resource is cleaned up. Default is OnExpiration and the cleanup time is after 1h.')
+param cleanupPreference string = 'OnExpiration'
+
+@description('Optional. The name of the storage account to use for the deployment script. An existing storage account is needed, if PrivateLink is going to be used for the deployment script.')
+param storageAccountName string = ''
+
+@description('Optional. The subnet id to use for the deployment script. An existing subnet is needed, if PrivateLink is going to be used for the deployment script.')
+param subnetId string = ''
+
+@description('Optional. Tags of the resource.')
+@metadata({
+  example: '''
+  {
+      "key1": "value1"
+      "key2": "value2"
+  }
+  '''
+})
+param tags object?
+
+// the following settings are only needed if the deployment script is using PrivateLink
+// they must only be added if they contain values. Conditional values for the properties does not work and ARM throws an error
+var storageSettings = !empty(storageAccountName)
```

Comment:
Instead of building the object yourself, it may make sense to re-use /reference the existing `deployment-script` module and have it do the heavy lifting: https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/resources/deployment-script/main.bicep